test/WebSites/InlineConstraintsWebSite/Startup.cs

```diff
@@ -12,6 +15,27 @@ public class Startup
         // Set up application services
         public void ConfigureServices(IServiceCollection services)
         {
+
+            services.Configure<RouteOptions>(
+                routeOptions => routeOptions.ConstraintMap.Add(
+                    "producttype",
+                    typeof(ItemTypeConstraint)));
+
+            services.Configure<RouteOptions>(
+                routeOptions => routeOptions.ConstraintMap.Add(
+                    "servicetype",
+                    typeof(ItemTypeConstraint)));
+
+            services.Configure<RouteOptions>(
+                routeOptions =>
+                {
+                    if (routeOptions.ConstraintMap.ContainsKey("servicetype"))
+                    {
+                        routeOptions.ConstraintMap["servicetype"] =
+                            typeof(ItemTypeUpperCaseConstraint);
+                    }
+                });
```

Comment:
would be cleaner to combine all of these in a statement lambda.

why do you need `if (routeOptions.ConstraintMap.ContainsKey("servicetype"))`?